shared/mail/mail.go

```diff
@@ -350,3 +354,15 @@ func SendMail(c smtpClient, mail mailData, date time.Time) error {
 
 	return nil
 }
+
+func (mail mailData) generateMessageID() string {
+	id := uuid.NewV4()
```

Comment:
Can we switch this to use the post id rather than a random id? I believe that'll help with proper threading, as per https://github.com/mattermost/mattermost-server/pull/18232.

Reply:
but where would i get the post id from?

Reply:
It would need to be passed in from the nearest context that contains it. You can see it done in the PR I hyperlinked.